src/os/ui/abstractmainctrl.js

```diff
@@ -236,11 +238,12 @@ os.ui.AbstractMainCtrl.prototype.initialize = function() {
     os.settings.set(['about', 'version'], this.scope['version']);
   }
 
-  // set the proxy url
-  os.net.ProxyHandler.PROXY_URL = /** @type {string} */ (os.settings.get(['proxyUrl'], os.net.ProxyHandler.PROXY_URL));
-
-  // set up the proxy the new way
-  os.net.ProxyHandler.configure(/** @type {?Object} */ (os.settings.get(['proxy'])));
+  // configure the proxy via settings & only add the handler if it's successful
+  if (os.net.ProxyHandler.configure(/** @type {?Object} */ (os.settings.get(['proxy'])))) {
```

Comment:
I don't think we need to get rid of the `proxyUrl` setting. I'm not sure it's used anywhere, but no reason we can't continue to support it. I would get `proxyUrl` and `proxy` from settings, and if either is defined go ahead and add the proxy handler and call the original two lines. Then `os.net.ProxyHandler.configure` doesn't need to change, which doesn't seem necessary.